packages/kbn-interpreter/common/lib/paths_registry.ts

```diff
@@ -17,52 +17,50 @@
  * under the License.
  */
 
-import clone from 'lodash.clone';
-
 class PathsRegistry {
+  private paths: Map<string, string[]>;
+
   constructor() {
-    this._paths = {};
+    this.paths = new Map();
   }
 
-  register = (type, paths) => {
-    if (!type) throw new Error(`Register requires a type`);
+  public register = (type: string, paths: any) => {
+    if (!type) {
+      throw new Error(`Register requires a type`);
+    }
     const lowerCaseType = type.toLowerCase();
 
     const pathArray = Array.isArray(paths) ? paths : [paths];
-    if (!this._paths[lowerCaseType]) {
-      this._paths[lowerCaseType] = [];
+    if (!this.paths.has(lowerCaseType)) {
+      this.paths.set(lowerCaseType, []);
     }
 
     pathArray.forEach(p => {
-      this._paths[lowerCaseType].push(p);
+      // @ts-ignore
+      this.paths.get(lowerCaseType).push(p);
     });
   };
 
-  registerAll = (paths) => {
-    Object.keys(paths).forEach(key => {
-      this.register(key, paths[key]);
+  public registerAll = (paths: any) => {
```

Comment:
To keep it consistent with the rest of the file, it's better to use `string` or `PathLike` as type

Reply:
it can be either string or string[] ...